### DISCUSSION TITLE
Why `BluetoothLeScanner` instead of `CompanionDeviceManager`?

### DISCUSSION THREAD

**Discussion by IgorGanapolsky** [HubbellCorp/SweetBlue#15] [Q&A] [answered]
Hello,
I am new to BLE development on Android.  From my reading of internet posts, it seems that `CompanionDeviceManager` is the recommended API for device discovery.  However, your repo makes use of `BluetoothLeScanner`.  I'd just like like to know the why behind this architectural decision?

Thanks,
Igor

**Comment by ryanhubbell** [comment, marked as answer]
CompanionDeviceManager is only available in API 26 and above. This library is actually a lot older than it may seem (it was developed long before API 26 was released). SweetBlue also supports API 18 and up, so in order to do that, you have to use BluetoothLeScanner.
